azerothcore/azerothcore-wotlk
# feat(Core/BG): revamp Battleground.QueueAnnouncer.Limit

## Changes Proposed:
-  Revamped the `Battleground.QueueAnnouncer.Limit.*` configuration
-  Instead of only being able to enable/disable it, added a `MinLevel` so it allows to enable only for that level and higher levels
- When limited, it announces only if one more player is needed to start the BG
-  At 80 it only limits RBG, at lower level only limits Warsong Gulch.

The config values are the following (let me know if something is not clear enough):

```
#
#    Battleground.QueueAnnouncer.Limit.MinLevel
#        Description: Limit the QueueAnnouncer starting from a certain level.
#                     When limited, it announces only if there are at least MinPlayers queued (see below)
#                     At 80 it only limits RBG, at lower level only limits Warsong Gulch.
#        Default:     0 - (Disabled, no limits)
#                     10 - (Enabled for all, because BGs start at 10)
#                     20 - (Enabled for 20 and higher)
#                     80 - (Enabled only for 80)

Battleground.QueueAnnouncer.Limit.MinLevel = 0

#
#    Battleground.QueueAnnouncer.Limit.MinPlayers
#        Description: When the Battleground.QueueAnnouncer.Limit.MinLevel limit is enabled (not 0)
#                     only show when at least MinPlayers are queued.
#        Default:     3 - (Show only when 3 or more players are queued)

Battleground.QueueAnnouncer.Limit.MinPlayers = 3

```

Basically:

```
- I want to limit the Queue announce messages
- I want to limit them only for Warsong Gulch (if level is < than 80) and RBG (at 80)
- I want to limit only starting from a certain level (e.g. if I set it to 30, I'm NOT limiting 10-19 and 20-29 but only 30-39 and higher)
- The limit is basically: show the announce only when there are AT LEAST Battleground.QueueAnnouncer.Limit.MinPlayers queued players
```

## Tests Performed:

- see comments below


## How to Test the Changes:

- Set `Battleground.QueueAnnouncer.SpamProtection.Delay = 1`
- Install https://github.com/azerothcore/mod-cfbg and set `CFBG.EvenTeams.Enabled = 1`
- Checkout https://github.com/azerothcore/mod-cfbg/pull/55 in the CFBG module (`cd modules/mod-cfbg; git checkout revamp-bg-queue-limit`)
- Play with the new configurations `Battleground.QueueAnnouncer.Limit.MinLevel` and `Battleground.QueueAnnouncer.Limit.MinPlayers` and make sure they work fine

- I hope the description and test instructions are clear enough, if not please let me know. Otherwise I'm going to assume it's clear for you how this is supposed to work and how to test it.


## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
